src/runtime/program/compute_budget/lib.zig

```diff
@@ -109,8 +111,169 @@ const ComputeBudgetInstruction = union(enum(u32)) {
 
     /// Set a specific transaction-wide account data size limit, in bytes, is allowed to load.
     set_loaded_accounts_data_size_limit: u32,
+
+    pub const Tag = @typeInfo(ComputeBudgetInstruction).@"union".tag_type.?;
+
+    const borsh = BorshStaticTaggedUnionHelper(ComputeBudgetInstruction);
+
+    pub const BorshDecodeError = error{
+        /// The raw data did not match the expected representation.
+        DecodeFailed,
+        /// `src` was too short.
+        TruncatedSrc,
+    };
+
+    /// Decodes the encoded ComputeBudgetInstruction, and the number of bytes it occupied in `src`.
+    pub fn borshDecodeSlice(
+        src: []const u8,
+    ) BorshDecodeError!struct { ComputeBudgetInstruction, borsh.EncodedSizeInt } {
+        if (src.len == 0) return error.TruncatedSrc;
+        const tag = borsh.tagDecode(src[0]) orelse return error.DecodeFailed;
+        const pl_size: borsh.EncodedSizeInt = borsh.payloadEncodedSize(tag);
+        const pl_bytes = src[1..];
+        if (pl_bytes.len < pl_size) return error.TruncatedSrc;
+        const cbi = borsh.decodeFromTagAndPayload(tag, pl_bytes[0..pl_size]) orelse
+            return error.DecodeFailed;
+        return .{ cbi, @sizeOf(u8) + pl_size };
+    }
 };
 
+// TODO: put this somewhere logical, maybe create a more general abstraction?
+fn BorshStaticTaggedUnionHelper(comptime Tagged: type) type {
+    const Tag, const has_payloads = switch (@typeInfo(Tagged)) {
+        .@"union" => |union_info| .{ union_info.tag_type.?, true },
+        .@"enum" => .{ Tagged, false },
```

Comment:
Why do we need to handle this case? It's not used for enums now, and if we need to serialize enums, isn't it already trivial without this helper? It might simplify things to deal with them separately.

Reply:
1ac743d1cb44b538039e16610bdb708314d6da4b